### ISSUE TITLE
New functions to template file output

### ISSUE THREAD

**Issue by holyhope** [jdkato/txtlint#474]
Hello,

Using the current `2.20.1` version, I try to generate a [valid junit.xml file](https://www.ibm.com/docs/en/developer-for-zos/15.0.0?topic=formats-junit-xml-format).

With the very basic functions available in the templating (with `--output=mytemplate.xml` flag), it is really pain in the ass to compute some basic metrics.
I miss-read the code and though that vale uses the very common [`sprig` library](http://masterminds.github.io/sprig/).
Since now, I found that it is not the case and it is very hard to compute data for a junit file from [`Data` structure](https://github.com/errata-ai/vale/blob/1cefe0bdf8fffb21fdab4ab5983cec7cc899092d/internal/cli/custom.go#L18-L21).

I suggest to add the sprig library to the template rendering.

What do you think?

**Comment by holyhope** [comment]
For your information, here is my first version of the template file:

```go-template
{{- $files := dict -}}
{{- $max_failure := 0 -}}
{{- $failures := 0 -}}
{{- range .Files -}}
    {{- $testcases := dict -}}
    {{- $total_for_file := 0 -}}
    {{- $path := .Path -}}
    {{- range .Alerts -}}
        {{- if .Hide -}}
            {{- continue -}}
        {{- end }}
        {{- $total_for_file = add1 $failures -}}
        {{- if not (hasKey $testcases .Check) -}}
            {{- $testcases = set $testcases .Check (list .) -}}
        {{- else -}}
            {{- $alerts := append (get $testcases .Check) . }}
            {{- $testcases = set $testcases .Check $alerts -}}
        {{- end -}}
    {{- end -}}
    {{- $files = set $files $path $testcases -}}
    {{- $failures = add $failures $total_for_file -}}
    {{- if lt $max_failure $total_for_file -}}
        {{- $max_failure = $total_for_file -}}
    {{- end -}}
{{- end -}}

<testsuites id="vale" name="vale" failures="{{- $failures -}}" tests="{{- mul $max_failure ($files | len) -}}">
    {{- range $path, $testcases := $files -}}
        <testsuite id="{{- $path -}}" name="{{- $path -}}" failures="{{- $testcases | len -}}" tests="{{- $max_failure -}}">
            {{- range $check, $alerts := $testcases -}}
                <testcase suite_name="{{ $check }}" name="{{- $check -}}" file="{{ $path }}">
                    {{- range $alerts -}}
                        <failure message="{{- printf "%s:%d" $path .Line -}} {{ .Description }}" type="{{- .Severity -}}">
{{- if empty .Action.Params | not -}}{{- .Action.Params | join ", " }}: {{ end -}}
{{- $path }}:{{ .Line }}:{{ .Span | first }}
{{ if empty .Match | not -}}{{ .Match }}: {{ end -}}
{{- .Message -}}
                        </failure>
                    {{- end -}}
                </testcase>
            {{- end -}}
        </testsuite>
    {{- end -}}
</testsuites>
```
